### ISSUE TITLE
gui: Incorrect application name when passing -regtest on Fedora

### ISSUE THREAD

**Issue by fanquake** [bitcoin/bitcoin#15079]
On Fedora 29.1-2, when running `bitcoin-qt -regtest`, the name of the application in the menu bar is shown as `Bitcoin-Qt-testnet` (the icon is the correct blue colour). i.e:
![fedora - notifications](https://user-images.githubusercontent.com/863730/50596726-23f17280-0ee0-11e9-9083-31c8baf62feb.png)

Noticed while testing #15000. When running just `bitcoin-qt`, or passing `-testnet` the menu name and icon are both correct.

**Comment by laanwj** [comment]
The same application name
```c++
#define QAPP_APP_NAME_TESTNET "Bitcoin-Qt-testnet"
```
is used for both regtest and testnet:
```c++
    {"test", QAPP_APP_NAME_TESTNET, 70, 30, QT_TRANSLATE_NOOP("SplashScreen", "[testnet]")},
    {"regtest", QAPP_APP_NAME_TESTNET, 160, 30, "[regtest]"}
```

**Comment by hebasto** [comment]
@laanwj 

> The same application name
> 
> ```c++
> #define QAPP_APP_NAME_TESTNET "Bitcoin-Qt-testnet"
> ```
> 
> is used for both regtest and testnet

Is it intended?

**Comment by laanwj** [comment]
No, to be honest I don't think anyone ever noticed this, or thought of this before. Regtest in GUI mode is very rarely used.

**Comment by jonasschnelli** [comment]
The application name itself is not a huge deal... but without the fix #15085, regtest and testnet GUI share the same QSettings which is strange that nobody detected this (including myself using GUI regtest a lot).

**Comment by MarcoFalke** [comment]
> regtest and testnet GUI share the same QSettings

Oh nice when this annoyance is finally fixed. Makes testing easier and less risky.
